tests/params.rs

```diff
@@ -0,0 +1,69 @@
+use http_types::{self, Method, Url};
+use tide::{self, Request, Response, Result, StatusCode};
+
+#[async_std::test]
+async fn test_param_invalid_type() {
+    async fn get_by_id(req: Request<()>) -> Result<Response> {
+        assert_eq!(
+            format!("{}", req.param::<i32>("id").err().unwrap()),
```

Comment:
This might read easier
```suggestion
            req.param::<i32>("id").unwrap_err().to_string(),
```